From c24185d069ecd59d89ca96e7268f8cc0965e3048 Mon Sep 17 00:00:00 2001
From: Michael Pauzner <mpauzner@habana.ai>
Date: Tue, 10 Dec 2024 16:29:51 +0200
Subject: [PATCH] [SW-207648] nic: add new nic ena set api

Change API.
Step 1, add temporary #define to pass multi-repository build.

Change-Id: I066a55d19e9bedbbd6f2c0488091d9485236e443
Signed-off-by: Michael Pauzner <mpauzner@habana.ai>
---

diff --git a/include/baremetal.h b/include/baremetal.h
index da9aad7..c7fd4b2 100644
--- a/include/baremetal.h
+++ b/include/baremetal.h
@@ -201,7 +201,8 @@
 void preboot_led_on(void);
 void preboot_pcie_enable(void);
 void preboot_pcie_exec_hw_wa(void);
-void preboot_nic_en(u32 nic_en_);
+int preboot_set_nic_en(u32 nic_en);
+#define preboot_nic_en(nic_en)		preboot_set_nic_en(nic_en)
 bool preboot_host_gpio_asserted(void);
 void preboot_sp_sram_init(void);
 void preboot_cond_arc0_reset(void);
diff --git a/lib/preboot/module_fwupd.c b/lib/preboot/module_fwupd.c
index edb4e58..ca26a09 100644
--- a/lib/preboot/module_fwupd.c
+++ b/lib/preboot/module_fwupd.c
@@ -638,7 +638,10 @@
 	__module_fwupd_run_fast(&i2c_state);
 }
 
-__weak void preboot_nic_en(u32 nic_en) {}
+__weak int preboot_set_nic_en(u32 nic_en)
+{
+	return 1;
+}
 
 static void module_fwupd_load_type_run(struct fwupd_state *state)
 {
